configure.ac

```diff
@@ -36,6 +36,31 @@ AS_IF([test "x$has_ncursesw" = "xyes"], [
   AC_SUBST([LIBS])
 ])
 
+AC_CANONICAL_HOST
+
+AC_ARG_ENABLE(thermal,
+  AS_HELP_STRING([--enable-thermal], [enable thremal event support (default is yes)]),,
+  [enable_thermal=yes]
+)
+
+PKG_CHECK_MODULES([LIBNL3], [libnl-3.0 libnl-genl-3.0], [has_libnl3=yes],
+  AC_MSG_NOTICE([libnl-3.0 is not available.])
+)
+
+AS_IF([test x"$host_cpu" = x"x86_64"],
+  AS_IF([test x"$enable_thermal" = x"yes"],
+    AS_IF([test x"$has_libnl3" = x"yes"], [
+      AC_MSG_NOTICE([support thermal event.])
+      support_thermal=yes;],
+      AC_MSG_NOTICE([no thermal event support without libnl-3.0.])),
+    AC_MSG_NOTICE([thermal event support is disabled with --enable-thermal=no])
+  )
+)
+AS_IF([test "x$support_thermal" = xyes], [
+  AC_DEFINE([HAVE_THERMAL], 1, [Build irqbalance to support thermal events])
+])
```

Comment:
This seems a bit backwards in terms of logic:

- PKG_CHECK_MODULES([LIBNL3]) only needs to be done if we select support thermal, so it should be moved inside the AS_IF([... support_thermal]) condition
- The 3 nested AS_IF conditions can just be a single test inside AS_IF([support thermal])
- You don't need both an enable_thermal and a support_thermal variable.
The whole thing can be simplified to the following pseudo code:

AC_ARG_ENABLE[--enable_thermal]
AS_IF([test <enable_thermal = yes> -a <host_cpu != x8664], [
     AC_MSG_ERROR([we don't support thermal events on non-intel platforms])], [])
AS_IF([test <enable_thermal = yes>], [
   PKG_CHECK_MODULES([LIBNL3]....)])

Reply:
Sorry, you also need to add the AC_DEFINE([HAVE_THERMAL]) in the action-if-found clause of PKG_CHECK_MODULES

Reply:
It sounds like you want the default of that option 'no', right?  Otherwise, it will always return error on non-x86 64-bit systems.

Reply:
No, I'm fine with thermal events being available by default, but you can do that with the AC_ARG_ENABLE macro - i.e. use the action-if-not-given clause to set thermal_enable=$enableval iff test host_cpu == x86_64

The rest is just to consolidate the validity checking and make it a bit more readable

Reply:
Okay, then the default is variable.  If user wants to set this on non-x86 systems, this error shows up; otherwise, will be smooth as the default is switched to no here.

I guess it should work. Let me check this under Qemu's ARM emulator.

Reply:
Okay, I found it works. You can see the changes from this line in the diff: https://github.com/Irqbalance/irqbalance/compare/3ec98d963fa3a483073e73a73998c2f317dabfc5..560291389e33db108d2fb6d954ef059f953c6e33#diff-49473dca262eeab3b4a43002adb08b4db31020d190caaad1594b47f1d5daa810R43